### PR TITLE
Re-factored the existing source plugins to be independent plugins

### PR DESCRIPTION
**Pull request by vishh**
All plugins now implement the sources/api.Source interface.

This lets us add arbitrary number of plugins in the future and ignore individual plugin failures.

### CODE REVIEW

**Comment by vishh** [comment]
cc @saad-ali

File: sources/kube_events.go [hunk 1]
```diff
@@ -175,3 +145,39 @@ func watchLoop(eventClient kubeclient.EventInterface, eventsChan chan<- eventsUp
...
+	}
+	glog.V(2).Info("Fetched list of events from the master")
+	glog.V(4).Infof("%v", events)
+
+	return api.AggregateData{Events: events}, nil
+}
+
+func (eventSource *eventsSourceImpl) DebugInfo() string {
+	desc := fmt.Sprintf("Source type: %s\n", kubeEventsSource)
+	// TODO: Add events specific ddebug information
```
Comment (rjnagal) [inline sources/kube_events.go]: s/ddebug/debug

File: sources/kube_events_test.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
+// Copyright 2014 Google Inc. All Rights Reserved.
```
Comment (rjnagal) [inline sources/kube_events_test.go]: 2015 in new files :)

**Comment by rjnagal** [comment]
LGTM
mostly nits.

File: sources/kube_events_test.go (hunk 2 above)
Comment (vishh) [inline sources/kube_events_test.go]: Done

**Comment by vishh** [comment]
@rjnagal: Thanks for the quick review. Addressed your comments.

**Comment by vishh** [comment]
@rjnagal: Build is green

**Pull request merged**